src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,1931 @@
+            using System;
+
+            C.M(new int[] { 7, 8, 9 });
+
+            static class C
+            {
+                public static void M(int[] arg) => ((Span<int>)arg).E();
+                public static void E(this Span<int> arg) => Console.Write(arg[1]);
+            }
+            """;
+
+        var expectedOutput = "8";
+
+        var comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular12);
+        var verifier = CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+        verifier.VerifyIL("C.M", """
+            {
+              // Code size       12 (0xc)
+              .maxstack  1
+              IL_0000:  ldarg.0
+              IL_0001:  call       "System.Span<int> System.Span<int>.op_Implicit(int[])"
+              IL_0006:  call       "void C.E(System.Span<int>)"
+              IL_000b:  ret
+            }
+            """);
+
+        var expectedIl = """
+            {
+              // Code size       12 (0xc)
+              .maxstack  1
+              IL_0000:  ldarg.0
+              IL_0001:  newobj     "System.Span<int>..ctor(int[])"
+              IL_0006:  call       "void C.E(System.Span<int>)"
+              IL_000b:  ret
+            }
+            """;
+
+        comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.RegularNext);
+        verifier = CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+        verifier.VerifyIL("C.M", expectedIl);
+
+        comp = CreateCompilationWithSpan(source);
+        verifier = CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+        verifier.VerifyIL("C.M", expectedIl);
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_ExtensionMethodReceiver_Opposite_Implicit(LanguageVersion langVersion)
+    {
+        var source = """
+            static class C
+            {
+                static void M(System.Span<int> arg) => arg.E();
+                static void E(this int[] arg) { }
+            }
+            """;
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (3,44): error CS1929: 'Span<int>' does not contain a definition for 'E' and the best extension method overload 'C.E(int[])' requires a receiver of type 'int[]'
+            //     static void M(System.Span<int> arg) => arg.E();
+            Diagnostic(ErrorCode.ERR_BadInstanceArgType, "arg").WithArguments("System.Span<int>", "E", "C.E(int[])", "int[]").WithLocation(3, 44));
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_ExtensionMethodReceiver_Opposite_Explicit(LanguageVersion langVersion)
+    {
+        var source = """
+            static class C
+            {
+                static void M(System.Span<int> arg) => ((int[])arg).E();
+                static void E(this int[] arg) { }
+            }
```

Comment:
>} [](http://example.com/codeflow?start=12&length=1)

Consider testing `ReadOnlySpan<int>` to `int[]` as well, perhaps in the same test.